CMakeLists.txt

```diff
@@ -20,7 +20,20 @@ cmake_minimum_required(VERSION 3.15)
 cmake_policy(SET CMP0069 NEW)  # LTO support
 cmake_policy(SET CMP0091 NEW)  # MSVC runtime support
 
-project(astcenc VERSION 3.0.0)
+project(astcencoder VERSION 3.0.0)
+
+option(UNIVERSAL_BUILD "Build multi-architecture universal binary [only supported on MacOS]" OFF)
+
+if (CMAKE_OSX_ARCHITECTURES)
+    message(STATUS "Building universal binary for ${CMAKE_OSX_ARCHITECTURES}")
+    set(UNIVERSAL_BUILD ON)
+endif()
+
+if (${UNIVERSAL_BUILD})
+    if (NOT CMAKE_OSX_ARCHITECTURES)
+        message(FATAL_ERROR "UNIVERSAL_BUILD requires -DCMAKE_OSX_ARCHITECTURES=\"arch1;arch2;...\"")
```

Comment:
Would be a more useful message if we actually list real architectures. Currently the user needs to know the right settings for x86 and Arm, the message should just tell them.

Reply:
Fixed in https://github.com/ARM-software/astc-encoder/pull/266/commits/1ef2b84a31901c36f85479527cfe48d8365546a1

Reply:
If we want to make it clear it's a hint, just stuff an "e.g." before "-DCMAKE...".

Reply:
This whole section is gone after I remove `${UNIVERSAL_BUILD}` as a user visible option. User needs to know how to set `CMAKE_OSX_ARCHITECTURES`. Will add docs to clarify it instead.